graphics.tex

```diff
@@ -330,4 +334,62 @@ \subsection*{VIC IV organization and memory access basics}
+set these two locations:
+
+\begin{screenoutput}
+50 POKE $D018, PEEK($D018) AND 247 OR 0 :REM BITMAP DATA STARTS AT $0000
+60 POKE $D018, PEEK($D018) AND 15 OR 32 :REM SCREEN RAM STARTS AT $0800
+\end{screenoutput}
+
+Of course you are free to choose where in memory you store your bitmap and color data. A detailed description of how to do this and how to configure it will follow later when we have switched from BASIC to machine code. 
+The configuration shown above is enough for now, ironically this is exactly the setting when you turn on your MEGA65 with the default bank 128 enabled. The final step is to poke our bitmap and color data into memory:
+
+\begin{screenoutput}
+70 POKE $0100, 0
+71 POKE $0101, 120
+72 POKE $0102, 204
+73 POKE $0103, 204
+74 POKE $0104, 252
+75 POKE $0105, 204
+76 POKE $0106, 204
+77 POKE $0107, 204
+80 POKE $0820, $A0
+\end{screenoutput}
+	
+In order to bring the already shown \textbf{A} from this figure
+
+\begin{center}
+	\begin{tabular}{|C{12pt}|C{12pt}C{12pt}C{12pt}C{12pt}C{12pt}C{12pt}C{12pt}C{12pt}|R{24pt}|R{24pt}|}
+	\hline
+		& 7 & 6 & 5& 4& 3&  2& 1& 0 & dec & hex \\
+	\hline
+	0 & \blkb & \blkb & \blkb & \blkb & \blkb & \blkb & \blkb & \blkb &   0 & 00 \\
+	1 & \blkb & \redb & \redb & \redb & \redb & \blkb & \blkb & \blkb & 120 & 78 \\
+	2 & \redb & \redb & \blkb & \blkb & \redb & \redb & \blkb & \blkb & 204 & CC \\
+	3 & \redb & \redb & \blkb & \blkb & \redb & \redb & \blkb & \blkb & 204 & CC \\
+	4 & \redb & \redb & \redb & \redb & \redb & \redb & \blkb & \blkb & 252 & FC \\
+	5 & \redb & \redb & \blkb & \blkb & \redb & \redb & \blkb & \blkb & 204 & CC \\
+	6 & \redb & \redb & \blkb & \blkb & \redb & \redb & \blkb & \blkb & 204 & CC \\
+	7 & \redb & \redb & \blkb & \blkb & \redb & \redb & \blkb & \blkb & 204 & CC \\
+	\hline
+	\end{tabular}
+	\end{center}
+
+to the screen, the values ​​must be placed in memory starting at \$0000. The color information in the screen RAM (please remember, it starts at \$0800) defines 
+the foreground and the background color of the corresponding block. In line 80 the value is split off into \textbf{A} for a light red foreground while the \textbf{0} is producing the wonderful black background. 
+The following screenshot shows the result, our "A" appears on the screen. But does it look good? 
+
+\includegraphics[width=\linewidth]{images/graphics/basic-01.png}
+	
+As you can probably guess, our BASIC implementation has many problems. It is difficult for us to move freely in memory. The "A" doesn't appear 
+at the very beginning of the screen. Look into the listing, poking starts at \$0100, not \$0000 as it should be. The reason is that the Zeropage starts at \$0000
+and storing graphic data there is not really clever. You see, we have to make a lot of compromises and face many challenges when we don't use BASIC's built-in graphics commands,
+but work directly in memory to generate graphics. 
+
+\subsection*{From BASIC to machine code}
```

Comment:
Probably remove this for now. I suspect that a chapter on graphics in general (i.e. not specific to BASIC graphics commands) will be mostly register and memory manipulation, in which case the differences between PEEK/POKE and lda/sta don't justify extra material. We should think about how we want to include memory mapping in this discussion.

(I intend to cover the BASIC graphics subsystems in a separate chapter and probably a separate book. If you want to contribute to that let me know, but I think that material should be separate from this chapter, because it's probably a best practice to not mix BASIC graphics commands and direct register manipulation.)

Reply:
Yeah ! Thanks for this valuable feedback, I will think about it, maybe i should skip that BASIC starter. Let me think about it. :-)